File: tools/dbench/answers/go_haiku/s3/r1_orders.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type Order struct {
	Id     int     `json:"id"`
	Amt    float64 `json:"amt"`
	Status string  `json:"status"`
}

func main() {
	buf, _ := os.ReadFile("orders.json")
	var orders []Order
	json.Unmarshal(buf, &orders)

	paidTotal := 0.0
	openCount := 0

	for _, ord := range orders {
		if ord.Status == "paid" {
			paidTotal += ord.Amt
		} else if ord.Status == "open" {
			openCount++
		}
	}

	fmt.Println(paidTotal)
	fmt.Println(openCount)
}

File: tools/dbench/answers/python_sonnet/s1/r1_orders.py
import json

with open("orders.json") as f:
    orders = json.load(f)

paid_total = sum(o["amt"] for o in orders if o["status"] == "paid")
open_count = sum(1 for o in orders if o["status"] == "open")
print(paid_total)
print(open_count)
